CMakeLists.txt

```diff
@@ -3,4 +3,9 @@ project(untitled)
 
 set(CMAKE_CXX_STANDARD 20)
 
-add_executable(untitled main.cpp)
\ No newline at end of file
+set(CMAKE_CXX_FLAGS "-pthread")
+
+set(SOURCE main.cpp Unit.cpp color.cpp Producer.cpp HashGen.cpp Consumer.cpp)
```

Comment:
You can use e.g. GLOB here not to forget to e.g. add newly created file
file(GLOB SOURCES *.cpp)
file(GLOB HEADERS *.h)
add_executable(${PROJECT_NAME}
${SOURCES}
${HEADERS}
)